packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx

```diff
@@ -0,0 +1,284 @@
+import { ChangeEvent, useMemo, useState, useCallback, useEffect } from 'react'
+import { Send as SendIcon } from 'react-feather'
+import { BigNumber } from 'bignumber.js'
+import { EthereumAddress } from 'wallet.ts'
+import {
+    Box,
+    Button,
+    createStyles,
+    IconButton,
+    InputAdornment,
+    makeStyles,
+    TextField,
+    Theme,
+    Typography,
+} from '@material-ui/core'
+import type { WalletRecord } from '../../../plugins/Wallet/database/types'
+import { ERC20TokenDetailed, EthereumTokenType, EtherTokenDetailed } from '../../../web3/types'
+import {
+    DashboardDialogCore,
+    DashboardDialogWrapper,
+    useSnackbarCallback,
+    WrappedDialogProps,
+} from '../DashboardDialogs/Base'
+import AbstractTab, { AbstractTabProps } from './AbstractTab'
+import { useI18N } from '../../../utils/i18n-next-ui'
+import { useTokenBalance } from '../../../web3/hooks/useTokenBalance'
+import { useCopyToClipboard } from 'react-use'
+import FileCopyOutlinedIcon from '@material-ui/icons/FileCopyOutlined'
+import { QRCode } from '../../../components/shared/qrcode'
+import { formatBalance, formatEthereumAddress } from '../../../plugins/Wallet/formatter'
+import { useTokenTransferCallback } from '../../../web3/hooks/useTokenTransferCallback'
+import { useRemoteControlledDialog } from '../../../utils/hooks/useRemoteControlledDialog'
+import { WalletMessages } from '../../../plugins/Wallet/messages'
+import { TransactionStateType } from '../../../web3/hooks/useTransactionState'
+
+interface WalletProps {
+    wallet: WalletRecord
+}
+
+//#region transfer tab
+const useTransferTabStyles = makeStyles((theme) =>
+    createStyles({
+        root: {
+            padding: theme.spacing(1),
+        },
+        button: {
+            marginTop: theme.spacing(3),
+        },
+    }),
+)
+
+interface TransferTabProps {
+    wallet: WalletRecord
+    token: ERC20TokenDetailed | EtherTokenDetailed
+    onClose: () => void
+}
+
+function TransferTab(props: TransferTabProps) {
+    const classes = useTransferTabStyles()
+    const { wallet, token, onClose } = props
+    const { t } = useI18N()
+
+    const [amount, setAmount] = useState('')
+    const [address, setAddress] = useState('')
+    const [memo, setMemo] = useState('')
+
+    // balance
+    const { value: tokenBalance = '0' } = useTokenBalance(token?.type ?? EthereumTokenType.Ether, token?.address ?? '')
+
+    const onChangeAmount = useCallback((ev: ChangeEvent<HTMLInputElement>) => {
+        const _amount = ev.currentTarget.value
+        if (_amount === '') setAmount('')
+        if (/^\d+[\.]?\d*$/.test(_amount)) setAmount(_amount)
```

Comment:
🤔Why not parse with Number.parseFloat then check if NaN?

Reply:
It prohibits users from entering non-numeric content.

Reply:
another question, why not use the number type input box?

Reply:
In this way, i can clearly input 0

Reply:
Cannot get value in this way `inputNode.value` if we use a number type input. @lelenei Take a look at `<TokenAmountPanel />` component, a bunch of helpful properties can be added to this component.

```jsx
inputProps: {
      autoComplete: 'off',
      autoCorrect: 'off',
      title: 'Token Amount',
      inputMode: 'decimal',
      min: 0,
      minLength: MIN_AMOUNT_LENGTH,
      maxLength: MAX_AMOUNT_LENGTH,
      pattern: '^[0-9]*[.,]?[0-9]*$',
      placeholder: '0.0',
      spellCheck: false,
}
```

Reply:
TokenAmoumntPanel type is text

Reply:
[
<img width="941" alt="image" src="https://user-images.githubusercontent.com/72531217/103633479-bb32aa00-4f80-11eb-9b83-a7a113cbfa34.png">
](url)